tools/highlight-files/highlight-files.ts

```diff
@@ -26,20 +27,40 @@ function getBazelActionArguments() {
 }
 
 if (require.main === module) {
-  // The script expects the bazel-bin path as first argument. All remaining arguments will be
-  // considered as markdown input files that need to be transformed.
-  const [bazelBinPath, ...inputFiles] = getBazelActionArguments();
-
-  // Walk through each input file and write transformed markdown output to the specified
-  // Bazel bin directory.
-  inputFiles.forEach(inputPath => {
-    const fileExtension = extname(inputPath).substring(1);
-    // Convert "my-component-example.ts" into "my-component-example-ts.html"
-    const baseOutputPath = inputPath.replace(`.${fileExtension}`, `-${fileExtension}.html`);
-    const outputPath = join(bazelBinPath, baseOutputPath);
-    const htmlOutput = highlightCodeBlock(readFileSync(inputPath, 'utf8'), fileExtension);
-
-    writeFileSync(outputPath, htmlOutput);
-  });
+    // The script expects the output directory as first argument. Second is the name of the
+    // package where this the highlight target is declared. All remaining arguments will be
+    // considered as markdown input files that need to be transformed.
+
+    const [outDir, packageName, ...inputFiles] = getBazelActionArguments();
+
+    // Walk through each input file and write transformed markdown output
+    // to the specified output directory.
+    inputFiles.forEach(execPath => {
+        // Compute a relative path from the package to the actual input file.
+        // e.g `src/components-examples/cdk/<..>/example.ts` becomes `cdk/<..>/example.ts`.
+        const basePath = relative(packageName, execPath);
+        const fileExtension = extname(basePath).substring(1);
+        const parsed = regionParser(readFileSync(execPath, 'utf8'), fileExtension);
+        for (let region in parsed['regions']) {
```

Comment:
Is there a reason you're using `['regions']` instead of `.regions`?

Also, since you're using both the key and value in this loop, can you do a `for..of` instead?
```js
for (const [regionName, regionSnippet] of Object.entries(parsed.regions)) {
  ...
}
```

Or is that not supported in this version of node?